add failure for signoffs from the doc example

robbkidd opened the pull request:
A signoff of "Signed-off-by: Julia Child <juliachild@chef.io>" is a sign that someone has followed the documentation a little too closely. We know Julia was not the author or committer of the commit, so mark it as
failed.

Alternatively—or as a follow-on to this—we could consider validating the sign-off information a bit more deeply. We could change the check to confirm that a commit's author or committer email appears in a sign-off entry.

robbkidd committed:
add failure for signoffs from the doc example

A signoff of "Signed-off-by: Julia Child <juliachild@chef.io>" is a sign
that someone has followed the documentation a little too closely. We
know Julia was not the author or committer of the commit, so mark it as
failed.

Signed-off-by: Robb Kidd <robb@thekidds.org>

thommay commented on src/lib/dcob/octoclient.rb:
```diff
@@ -59,13 +62,14 @@ def dco_check_success(repository_id, commit_sha)
 
-    def dco_check_failure(repository_id, commit_sha)
+    def dco_check_failure(repository_id, commit_sha,
+                          message = "This commit does not have a DCO Signed-off-by")
```
i lean towards keyword arguments for things like this, but 👍 either way.

robbkidd replied:
I was sensing that keyword arguments would be better here, too. I'll adjust.

robbkidd committed:
update dco_check_failure to use keyword arguments

With the addition of a failure message, keyword arguments will make
things clearer at call points.

Signed-off-by: Robb Kidd <robb@thekidds.org>

robbkidd committed:
update dco_check_success to use keyward arguments

Having changed dco_check_failure to accept a customizable failure
message, it became apparent that dco_check_success could follow the same
pattern and obviate the need for a separate method for obvious fixes.

Signed-off-by: Robb Kidd <robb@thekidds.org>

robbkidd committed:
move puts loggish statements into the check methods

Signed-off-by: Robb Kidd <robb@thekidds.org>

robbkidd committed:
set a rubocop config; do the frozen string thing

See
http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/FrozenStringLiteralComment

Signed-off-by: Robb Kidd <robb@thekidds.org>
